### PR TITLE
[Hub Generated] Review request for Microsoft.KeyVault to add version stable/2019-09-01

### PR DESCRIPTION
**Pull request by jiacheng-L**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/jiacheng-L/azure-rest-api-specs/dev-keyvault-Microsoft.KeyVault-2019-09-01-LintingFix...master/).

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [x] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “**WaitForARMFeedback**” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://msazure.visualstudio.com/One/_workitems/create/Scenario?templateId=d1c215ec-5d6a-4cb3-a193-95ddd0a1ba96&ownerId=85bb92fb-21fe-41ca-9309-47f2c5f57515). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by raych1** [comment]
> Hi @jiacheng-L, one or multiple breaking change(s) is detected in your PR. Please check out the breaking change(s), and provide business justification in the PR comment and @ PR assignee why you must have these change(s), and how external customer impact can be mitigated. Please ensure to follow [breaking change policy](https://aka.ms/AzBreakingChangesPolicy) to request breaking change review and approval before proceeding swagger PR review.
> **Action**: To initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).
> If you want to know the production traffic statistic, please see [ARM Traffic statistic](https://portal.azure-devex-tools.com/api/pr/query-traffic?pr=Azure/azure-rest-api-specs/12475&days=5).
> If you think it is false positive breaking change, please provide the reasons in the PR comment, report to Swagger Tooling Team via https://aka.ms/swaggerfeedback.

@jiacheng-L , can you take above **action** for the breaking change review request?

**Review by JeffreyRichter — APPROVED** [review]

File: specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2019-09-01/providers.json
Comment (heaths) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2019-09-01/providers.json:165]: Why is the actual error nested? Is this common of mgmt plane?

File: specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2019-09-01/providers.json
Comment (heaths) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2019-09-01/providers.json:171]: This is the same as `CloudError`. Does a new model need to be created? Or is the thought they could diverge at some point?

**Review by heaths — APPROVED** [review]

File: specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2019-09-01/providers.json
Comment (jiacheng-L) [reply specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2019-09-01/providers.json:165]: I believed so. Both Key.json and KeyVault.json use the same model.

File: specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2019-09-01/providers.json
Comment (jiacheng-L) [reply specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2019-09-01/providers.json:171]: They can be the same. So in the preview version, I put the default error model into the common file.

**Comment by raych1** [comment]
@ArcturusZhang @msyyc , can you double check the SDK breaking change? Thanks.

File: specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2019-09-01/providers.json
Comment (jiacheng-L) [reply specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2019-09-01/providers.json]: They are the same. I updated it to CloudError, as well as the one in secrets.json, which I defined in the last PR

Comment (jiacheng-L) [reply specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2019-09-01/providers.json]: @heaths They are the same. I updated it to CloudError, as well as the one in secrets.json, which I defined in the last PR

**Comment by ArcturusZhang** [comment]
Some of the breaking changes are introduced by #11607 which are not acceptable and wrong. I am looking at the root cause of this issue. Please temporarily do not merge this PR. Thank you

**Comment by jiacheng-L** [comment]
> 
> 
> Some of the breaking changes are introduced by #11607 which are not acceptable and wrong. I am looking at the root cause of this issue. Please temporarily do not merge this PR. Thank you

If you are talking about the .net breaking change, it may probably because I added the Error response block in that PR. Error response model can not covert to the previous 'Vault' result object. Therefore I may need to submit a request to update the .net SDK

**Comment by jiacheng-L** [comment]
> 
> 
> > Some of the breaking changes are introduced by #11607 which are not acceptable and wrong. I am looking at the root cause of this issue. Please temporarily do not merge this PR. Thank you
> 
> If you are talking about the .net breaking change, it may probably because I added the Error response block in that PR. Error response model can not covert to the previous 'Vault' result object. Therefore I may need to submit a request to update the .net SDK

But for Go and Python, I don't have any ideas why those Breaking Change tag would be added

**Comment by ArcturusZhang** [comment]
@jiacheng-L the swaggers with the new added response codes generate go code that cannot work properly, because previous the go generator cannot handle the extension `x-ms-error-response`.
I am making a fix. When the fix is done (very soon), the breaking change (at least the response code part) should go away

**Review by jlichwa — APPROVED** [review]

**Comment by raych1** [comment]
> @jiacheng-L the swaggers with the new added response codes generate go code that cannot work properly, because previous the go generator cannot handle the extension `x-ms-error-response`.
> I am making a fix. When the fix is done (very soon), the breaking change (at least the response code part) should go away

@ArcturusZhang , let me know once your fix is merged and I will remove `DONOTMERGE` label.

**Comment by ArcturusZhang** [comment]
Hi @raych1 after a rethinking, I suppose it does not make sense blocking this PR from merging by an issue caused by a merged PR. Please merge this when it is ready as usual. Therefore I removed the `DoNotMerge` label on this PR.

And BTW the fix is ready, but we need to test it.

**Comment by raych1** [comment]
> Hi @raych1 after a rethinking, I suppose it does not make sense blocking this PR from merging by an issue caused by a merged PR. Please merge this when it is ready as usual. Therefore I removed the `DoNotMerge` label on this PR.
> 
> And BTW the fix is ready, but we need to test it.

Thanks @ArcturusZhang , it makes sense. @msyyc, can you please check if this PR has python sdk breaking change?

**Comment by msyyc** [comment]
For python, the PR is ok

**Pull request merged**
